src/shared/logs-show.c

```diff
@@ -874,10 +872,9 @@ static int output_json(
                         size_t m;
                         unsigned u;
 
-                        /* We already printed the boot id, from the data in
+                        /* We already printed the boot id from the data in
                          * the header, hence let's suppress it here */
-                        if (length >= 9 &&
-                            memcmp(data, "_BOOT_ID=", 9) == 0)
+                        if (length >= 9 && memcmp(data, "_BOOT_ID=", 9) == 0)
```

Comment:
might be time to add some macro for this actually, "memory_startswith()" or so, or maybe "startswith_n()"?

```c
static inline void *memory_startswith(const void *p, size_t sz, const char *token) {
        size_t n;
        n = strlen(token);
        if (sz < n)
                return NULL;
        if (memcmp(p, token, n) == 0)
                return (uint8_t*) p + n;
        return NULL;
}
```

or so? (but it's definitely material for a later PR)

Reply:
(I am working on adding a helper for this now, btw, for an independent PR)

Reply:
i posted that pr as #9131 btw